### PR TITLE
[GPU] update weights_layout for GroupConv 1d spatial

### PR DESCRIPTION
**Pull request by sungeunk**
### Tickets:
 - 108266

### CODE REVIEW

**Comment by hyunback** [comment]
@sungeunk  Without any comments, it can be difficult to know later, Could you update to briefly what's PR do?

File: src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.hpp [hunk 1]
```diff
@@ -21,10 +21,19 @@ static std::shared_ptr<dnnl::convolution_forward::primitive_desc> get_convolutio
     dnnl::memory::dims pad_l(prim->pad.begin(), prim->pad.end());
     dnnl::memory::dims pad_r(prim->pad.begin(), prim->pad.end());
 
+    auto grouped_weights = format::is_grouped(weights_layout.format) || prim->grouped_weights_shape;
+    if (grouped_weights && (input_layout.get_rank() == weights_layout.get_rank())) {
+        auto tensor = weights_layout.get_tensor();
+        if (tensor.spatial[0] == 1 && tensor.spatial[1] != 1) {
```
Comment (hyunback) [inline src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.hpp:27]: Two questions,
1. How about the opposite case? (For example spatial[0] != 1 && spatial[1] == 1)
2. What about more than 4 dims?

**Comment by isanghao** [comment]
Performance was good. --this ../log/pr/2023-04-21--23-52.report --ref ../log/2023-04-21--20-18.report

Please check Hyunback's comment.

File: src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.hpp (hunk 1 above)
Comment (sungeunk) [reply src/plugins/intel_gpu/src/graph/impls/onednn/convolution_onednn.hpp:39]: XY of weight tensor is changed by ngraph or something.
I'm not sure there is the case that used Y not X. If it is, this logic needs to know why XY values are swapped.

Tensor of group weight will be increased 1 dim more than input data.

**Review by hyunback — APPROVED** [review]

**Review by isanghao — APPROVED** [review]
Looks good to me. Let me check the performance.

**Comment by isanghao** [comment]
no perf regression --this /home/mingyuki/log/pr/2023-04-25--14-50.report --ref /home/mingyuki/log/2023-04-24--19-24.report

**Pull request merged**